src/AppInstallerRepositoryCore/Rest/Schema/1_0/Json/SearchResponseDeserializer.cpp

```diff
@@ -34,12 +34,12 @@ namespace AppInstaller::Repository::Rest::Schema::V1_0::Json
     std::optional<IRestClient::SearchResult> SearchResponseDeserializer::DeserializeSearchResult(const web::json::value& searchResponseObject) const
     {
         // Make search result from json output.
+        IRestClient::SearchResult result;
         if (searchResponseObject.is_null())
         {
-            return {};
+            return result;

```

Comment:
>return result; [](start = 12, length = 14)

Curious why the change here? The previous logic of returning nullopt seems reasonable

Reply:
So, if you see below, I return empty set of results if there is a missing data array in the response. This is as good as saying, if there is no response content, then there are no search results. So instead of throwing an expection for a similar case (where the response is null), I changed it to return empty results

---
In reply to: [612849068](https://github.com/microsoft/winget-cli/pull/861#discussion_r612849068) [](ancestors = 612849068)

Reply:
Per offline discussion, as we are specifying MIME type as application/json, a valid empty result will be {} instead of "". Reverted the change and added this check in other deserializers.

---
In reply to: [613481181](https://github.com/microsoft/winget-cli/pull/861#discussion_r613481181) [](ancestors = 613481181,612849068)